### DISCUSSION TITLE
Check textobject under cursor

### DISCUSSION THREAD

**Discussion by phgz** [nvim-treesitter/nvim-treesitter-textobjects#353] [Q&A] [answered]
Hello,

Sorry if that has already been asked, but I was wondering how to retrieve the text object under the cursor (from the ones defined as built-in in the README).

I am trying to make a dynamic **swap** which swaps with previous or next node which is the same as the one under the cursor. At the moment, I do a mapping as such (only works for python):
```lua
local mapping = {
...
    if_statement = "conditional",
    for_statement = "loop",
    class_definition = "class",
    argument_list = "parameter",
    parameters = "parameter",
...
}
```
I get the `node:type()` (or `node:parent()` if no match) to retrieve the key, and map to the textobject to finally call, for example `require("nvim-treesitter.textobjects.swap").swap_previous("@conditional.outer")`.

Is there a way to retrieve the "mapping list" for each language? I saw in "queries" that there seems to be what I'm looking for, but I don't know how to use it.

Do you have a hint of what to do or where to look?

Thanks again!

**Comment by kiyoon** [comment, marked as answer]
Hi, I saw `siblings-swap.nvim` and `iswap.nvim` which presumably do this thing. You can have a look at their implementation

**Reply by phgz** [reply]
Will do!
